### PR TITLE
[Enhancement] Add trace log for rowset shared_ptr ref/deref

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Relates to #5409

## Problem Summary(Required) ：

Add some trace log to debug #5409

### CODE REVIEW

File: be/src/exec/vectorized/olap_scan_node.cpp [hunk 1]
```diff
@@ -562,8 +567,11 @@ Status OlapScanNode::_capture_tablet_rowsets() {
             std::shared_lock l(tablet->get_header_lock());
             RETURN_IF_ERROR(tablet->capture_consistent_rowsets(Version(0, version), &_tablet_rowsets[i]));
         }
+        StringAppendF(&_opened_tablet_ids, "%ld,", tablet_id);
+    }
+    if (!_opened_tablet_ids.empty()) {
+        LOG(INFO) << "open rowsets for " << _opened_tablet_ids;
```
Comment (imay) [inline be/src/exec/vectorized/olap_scan_node.cpp:573]: This will cause too many logs, better to use VLOG

Comment (decster) [reply be/src/exec/vectorized/olap_scan_node.cpp:573]: will not merge, just run in daily

File: be/src/exec/pipeline/scan/olap_scan_operator.cpp [hunk 2]
```diff
@@ -64,8 +65,16 @@ void OlapScanOperator::do_close(RuntimeState* state) {
     _ctx->unref(state);
 }
 
+OlapScanOperator::~OlapScanOperator() {
+    if (!_opened_tablet_ids.empty()) {
+        VLOG(1) << "close rowsets for " << _opened_tablet_ids << " query:" << print_id(_queryid);
+        _opened_tablet_ids.clear();
```
Comment (mofeiatwork) [inline be/src/exec/pipeline/scan/olap_scan_operator.cpp:71]: Why clear it rather than release the memory directly?

Comment (decster) [reply be/src/exec/pipeline/scan/olap_scan_operator.cpp:71]: This pr only adds log, not changing code behavior,  `~OlapScanOperator` destructor should do the job, so I think `_opened_tablet_ids.clear();` is also unnecessary

**Pull request closed without merging**